src/target/stm32f1.c

```diff
@@ -486,9 +499,14 @@ static bool stm32f1_cmd_option(target *t, int argc, const char **argv)
 	target_mem_write32(t, FLASH_OPTKEYR, KEY1);
 	target_mem_write32(t, FLASH_OPTKEYR, KEY2);
 
-	if ((argc == 2) && !strcmp(argv[1], "erase")) {
+	if ((argc == 2) && strcmp(argv[1], "erase") == 0) {
```

Comment:
the ()'s around the argc check are redundant and suppress warnings about accidental assignment, please remove them as part of this change